### ISSUE TITLE
Accessibility problem with pagination component

### ISSUE THREAD

**Issue by gestchild** [wellcomecollection/wellcomecollection.org#10470]
We have a pagination component that allows a user to navigate directly to a particular page by entering the desired page number in an input field: 

<img width="325" alt="Screenshot 2023-12-01 at 11 09 26" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/6051896/52be4ed9-4cf7-4034-bbd8-01d1180a9b54">

However, there is no submit button for the input and the interaction relies on using the enter key to change the page.

This causes an accessibility issue, that is being flagged on our [pa11y dashboard](https://dash.wellcomecollection.org/pa11y/)
"error: This form does not contain a submit button, which creates issues for those who cannot submit the form using the keyboard."

In order to fix this we'd need to provide a submit button to the user. 
~N.B. doing this could also make this functionality available to users without javascript enabled~ - thinking about it we should be able to do this anyway

**Comment by rcantin-w** [comment]
Won't fix in favour of #10684 - will close
